sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -84,8 +99,21 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
   private async getToken(options: GetTokenOptions): Promise<string | undefined> {
     let accessToken = this.tokenCache.getCachedToken();
     if (accessToken === undefined) {
-      accessToken = (await this.credential.getToken(this.scopes, options)) || undefined;
-      this.tokenCache.setCachedToken(accessToken);
+      // Waiting for the next refresh only if the cache is unable to retrieve the access token,
+      // which means that it has expired, or it has never been set.
+      const refreshPromise = this.tokenRefresher.refresh(options);
+      if (refreshPromise !== null) {
+        accessToken = await refreshPromise;
+      }
+    } else {
+      // If we still have a cached access token,
+      // And any other time related conditionals have been reached based on the tokenRefresher class,
+      // then attempt to refresh without waiting.
+      if (this.tokenRefresher.ready()) {
+        this.tokenRefresher.refresh(options).then((accessToken: AccessToken | undefined) => {
+          this.tokenCache.setCachedToken(accessToken);
+        });
+      }
```

Comment:
I think I'd rather replace this bit of logic with a call to a private function like

```suggestion
      this.updateTokenIfNeeded();
```

```ts
private async updateTokenIfNeeded(options: GetTokenOptions): Promise<void> {
   if (this.tokenRefresher.ready()) {
      const accessToken = await this.tokenRefresher.refresh(options);
      this.tokenCache.setCachedToken(accessToken);
   }
}
```

Reply:
Done!